quodlibet/quodlibet/ext/events/synchronize_to_device.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Jan Korte
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+import os
+import string
+from pathlib import Path
+
+from gi.repository import Gtk, GLib
+from quodlibet import _, app
+from quodlibet import config
+from quodlibet import get_user_dir
+from quodlibet import qltk
+from quodlibet.plugins import PluginConfigMixin
+from quodlibet.plugins.events import EventPlugin
+from quodlibet.qltk.ccb import ConfigCheckButton
+from quodlibet.query import Query
+
+from shutil import copyfile
+
+
+class SyncToDevice(EventPlugin, PluginConfigMixin):
+    PLUGIN_ID = "synchronize_to_device"
+    PLUGIN_NAME = _("Synchronize to Device")
+    PLUGIN_DESC = _(
+        "Synchronizes all songs from the selected saved searches with the "
+        "specified folder. All songs in that folder, which are not in the "
+        "saved searches, will be deleted.")
+    config_path_key = __name__ + '_path'
+
+    def PluginPreferences(self, parent):
+        vbox = Gtk.VBox(spacing=6)
+
+        query_path = os.path.join(get_user_dir(), 'lists', 'queries.saved')
+        try:
+            with open(query_path, 'r', encoding="utf-8") as query_file:
+                if not query_file.read(1):
+                    raise FileNotFoundError
```

Comment:
This check prevents reading the first character of the `query_file`, and the query at the top of the file will be parsed incorrectly.

Example: with the current code, a query of `title=Sun` at the top of the file will be parsed as `itle=Sun`.

To show the "No saved searches yet" message when the file is empty, I suggest the following:

1. Remove this `if` statement:
    ```suggestion
    ```

1. _(Optional)_ Create a custom exception class to clarify why we're raising an exception:

    ```python
    class NoSavedQueriesError(Exception):
        pass
    ```

1. Add an `else` clause after reading the `query_file`:

    ```python
    queries = {}
    for query_string in query_file:
        name = next(query_file).strip()
        queries[name] = Query(query_string.strip())
    else:
        # query_file is empty
        raise NoSavedQueriesError
    ```

    or add a new `if` statement after the `for` loop:

    ```python
    if not queries:
        raise NoSavedQueriesError
    ```

1. _(Optional)_ If using the custom exception class, update the `except` clause at the bottom of the file:

    ```python
    except NoSavedQueriesError:
    ```